test/e2e/kubevirt.go

```diff
@@ -1602,27 +1695,61 @@ runcmd:
 				})
 
 				const secondaryInterfaceName = "eth1"
-				Expect(setupUnderlay(nodes, secondaryBridge, secondaryInterfaceName, netConfig)).To(Succeed())
+				Expect(setupUnderlay(nodes, secondaryBridge, secondaryInterfaceName, networkName, 0 /*vlanID*/)).To(Succeed())
+			}
+			createCUDN(cudn)
+
+			if td.ingress == "routed" {
+				createRA(&rav1.RouteAdvertisements{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: cudn.Name,
+					},
+					Spec: rav1.RouteAdvertisementsSpec{
+						Advertisements: []rav1.AdvertisementType{rav1.PodNetwork},
+						NetworkSelectors: crdtypes.NetworkSelectors{{
+							NetworkSelectionType: crdtypes.ClusterUserDefinedNetworks,
+							ClusterUserDefinedNetworkSelector: &crdtypes.ClusterUserDefinedNetworkSelector{
+								NetworkSelector: metav1.LabelSelector{
+									MatchLabels: map[string]string{"name": cudn.Name},
+								},
+							},
+						}},
+					},
+				})
 			}
 
-			By("Creating NetworkAttachmentDefinition")
-			nad = generateNAD(netConfig)
-			Expect(crClient.Create(context.Background(), nad)).To(Succeed())
 			workerNodeList, err := fr.ClientSet.CoreV1().Nodes().List(context.Background(), metav1.ListOptions{LabelSelector: labels.FormatLabels(map[string]string{"node-role.kubernetes.io/worker": ""})})
 			Expect(err).NotTo(HaveOccurred())
 			selectedNodes = workerNodeList.Items
 			Expect(selectedNodes).NotTo(BeEmpty())
 
-			iperfServerTestPods, err = createIperfServerPods(selectedNodes, netConfig, []string{})
+			iperfServerTestPods, err = createIperfServerPods(selectedNodes, cudn.Name, td.role, []string{})
 			Expect(err).NotTo(HaveOccurred())
 
+			routerContainerName := "frr"
```

Comment:
can't this be a const ?

Reply:
I am going to just remove it and inline it at function call, we are just using it at one place.